node/src/blockchain/blockchain_interface.rs

```diff
@@ -988,6 +985,7 @@ mod tests {
         let subject =
             BlockchainInterfaceWeb3::new(transport, event_loop_handle, TEST_DEFAULT_CHAIN);
 
+        let end_block_nbr = 1024u64;
```

Comment:
I'm puzzled. Obviously, `end_block_nbr` has been part of this test since before these changes, because it's used in the last assert. But I don't see where it was set before this line (absent until these changes) came into existence. Do you know where it came from before?

Reply:
I may have refactored to allow the test to reuse the value in more than one place... part of the call to `subject.retrieve_transactions()` as `BlockNumber::Number(end_block_nbr.into())` and the value appears as part of the request payload `"fromBlock":"0x2a","toBlock":"0x400"` though not as obvious.

Reply:
Never mind: I see it now. `end_block_nbr` used to be just `42`. I swear the last time I looked at this code, the "`1 + end_block_nbr`" in the last assertion wasn't green. But it is now, and it's obviously been changed from "`42 + 1`", so it all makes sense now. Never mind. Sorry.